great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1266,6 +1266,7 @@ def get_batch_data_and_markers(
                 execution_engine=self,
                 selectable=selectable,
                 create_temp_table=create_temp_table,
+                temp_table_schema_name=temp_table_schema_name,
```

Comment:
sqlalchemy_batch_data.py uses this variable in `self._generate_selectable_from_selectable`.
Method `_generate_selectable_from_selectable` uses `self._create_temporary_table` passing `temp_table_schema_name`. 

I am using Snowflake and `_create_temporary_table` and the following code allows me with the change to create the temporary table in another database / schema.

For example, setting temp_table_schema_name to `PUBLIC.TEST` will create the temporary table in database public and schema test.

Reply:
I am passing `temp_table_schema_name` to the checkpoint class as `validations` batch_request with `batch_spec_passthrough`.